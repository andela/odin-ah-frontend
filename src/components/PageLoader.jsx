import React from 'react';
import PropTypes from 'prop-types';

export default function PageLoader({ text }) {
  return (
    <div id="loader" className="center-in-page loading-bg">
      <div>
        <div className="lds-dual-ring"/>
      </div>
      <p>{text}</p>
    </div>
  );
}


PageLoader.propTypes = {
  text: PropTypes.string,
};

PageLoader.defaultProps = {
  text: 'Loading'
};
