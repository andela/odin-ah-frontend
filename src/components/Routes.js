import React from 'react';
import { Route, Switch } from 'react-router-dom';
import Home from './Home/HomeContainer';
import Toast from './notification/toast';
import Modal from './modal';
import CreateArticle from './article/createArticle/createArticle';

export default function Root() {
  return (
    <div>
      <Switch>
        <Route exact path="/" component={Home} />
        <Route exact path='/article/add' component={CreateArticle} />
      </Switch>

      <Toast/>
      <Modal/>
    </div>
  );
}
