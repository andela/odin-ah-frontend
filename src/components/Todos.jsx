import React from 'react';
import { connect } from 'react-redux';
import { simpleAction } from '../redux/actions/todos/todos';

export function Todos() {
  return (
    <div>
      <h1>Hello I have a state in my redux store</h1>
    </div>
  );
}

export default connect(null, simpleAction)(Todos);
