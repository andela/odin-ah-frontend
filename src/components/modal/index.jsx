import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { dismissModal } from '../../redux/actions/modal';

export class Modal extends Component {
  dismissBtn = () => {
    this.props.dismissModal();
  };

  render() {
    const Content = this.props.content.Component;
    const { props } = this.props.content;
    const { show } = this.props;
    if (!Content) return null;
    return (
        <div className={`modal ${(show) ? 'is-active' : ''}`}>
          <div className="modal-background"/>
          <div className="modal-content content">
            <Content {...props}/>
          </div>
          <button onClick={this.dismissBtn} className="modal-close is-large" aria-label="close"/>
        </div>
    );
  }
}

Modal.propTypes = {
  content: PropTypes.shape({
    Component: PropTypes.any,
    props: PropTypes.any,
  }),
  show: PropTypes.bool,
  dismissModal: PropTypes.func.isRequired,
};

Modal.defaultProps = {
  content: {
    props: {}
  },
  show: true,
};

const mapStateToProps = state => ({
  show: state.modal.show,
  content: state.modal.content,
});

export default connect(mapStateToProps, { dismissModal })(Modal);
