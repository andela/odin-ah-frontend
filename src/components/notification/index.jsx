import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './index.scss';

export class Notification extends Component {
  static createMarkup(text) {
    return { __html: text };
  }

  constructor(props) {
    super(props);
    this.toast = React.createRef();
    this.type = {
      success: 'is-success',
      error: 'is-danger',
      info: 'is-info',
      warning: 'is-warning'
    };
    this.timeout = null;
  }

  // eslint-disable-next-line camelcase
  UNSAFE_componentWillReceiveProps(newProps) {
    this.setTimeout(newProps.show);
  }

  setTimeout(show) {
    if (this.timeout) {
      clearTimeout(this.timeout);
      this.timeout = null;
    }
    if (show && this.props.autoDismiss) {
      const { timeout } = this.props.data;
      this.timeout = setTimeout(() => {
        this.props.dismiss();
      }, timeout);
    }
  }

  render() {
    const { data, show } = this.props;
    const type = this.type[data.type];
    const hide = (!show) ? 'dismiss' : 'show';
    return (
      <div ref={this.toast} className={`${this.props.mode} notification ${type} ${hide}`}>
        <button onClick={this.props.dismiss} className="delete"/>
        <p dangerouslySetInnerHTML={Notification.createMarkup(data.text)}/>
      </div>);
  }
}

export const basePropTypes = {
  data: PropTypes.shape({
    id: PropTypes.number,
    text: PropTypes.string,
    timeout: PropTypes.number,
    type: PropTypes.oneOf(['success', 'error', 'info', 'warning']),
  }).isRequired,
  show: PropTypes.bool,
  mode: PropTypes.oneOf(['toast', 'alert']),
};

export const defaultNotificationData = {
  text: '',
  type: 'success',
  timeout: 8000
};

Notification.propTypes = {
  ...basePropTypes,
  autoDismiss: PropTypes.bool,
  dismiss: PropTypes.func,
};

Notification.defaultProps = {
  show: false,
  data: defaultNotificationData,
  autoDismiss: true,
};

export default Notification;
