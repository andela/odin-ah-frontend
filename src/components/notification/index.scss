.alert {

}
.toast {
  visibility: hidden;
  width: 250px;
  max-width: 300px;
  right: 0;
  margin-right: 16px;
  top: 100px;
  font-size: 16px;
  display: flex;
  position: fixed;
  box-shadow: 1px 1px 5px #ccc;
  z-index: 2000;
}

.toast.dismiss {
  -webkit-animation: fadeout 0.5s;
  animation: fadeout 0.5s;
}

.toast.show {
  visibility: visible;
  -webkit-animation: fadein 0.5s;
  animation: fadein 0.5s;
}

.alert.dismiss {
  display: none;
}

.alert.show {
  display: block;
}

.toast .title {
  font-size: 16px;
  font-weight: bold;
}
.toast .body {
  font-size: 24px;
}

@-webkit-keyframes fadein {
  from {
    right: -300px;
    opacity: 0;
  }
  to {
    right: 0;
    opacity: 1;
  }
}

@keyframes fadein {
  from {
    right: -300px;
    opacity: 0;
  }
  to {
    right: 0;
    opacity: 1;
  }
}

@-webkit-keyframes fadeout {
  from {
    right: 0;
    opacity: 1;
  }
  to {
    right: -300px;
    opacity: 0;
  }
}

@keyframes fadeout {
  from {
    right: 0;
    opacity: 1;
  }
  to {
    right: -300px;
    opacity: 0;
  }
}
