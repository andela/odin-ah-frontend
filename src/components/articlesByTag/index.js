import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import ArticleListView from '../articleList/ArticleListView';
import { getArticlesByTag } from '../../redux/actions/articles/articles';
import { logout } from '../../redux/actions/auth/login';
import PageLoader from '../PageLoader';
import './style.scss';
import Alert from '../notification/alert';

export class ArticlesByTag extends Component {
  // eslint-disable-next-line camelcase
  UNSAFE_componentWillMount() {
    const { match } = this.props;
    const { name } = match.params;
    this.props.getArticlesByTag(name);
  }

  render() {
    const { articles, match } = this.props;
    const { name } = match.params;
    return (
      <React.Fragment>
        <Alert/>
          {this.props.loading && <PageLoader text="Loading..." />}
          {!this.props.loading && (<section className="section">
          <div className='container article-tag '>
            <div className='columns'>
              <div className='column'>
                <div className='article-tag__box'>
                  <h2 className="article-tag__title">Tagged <span className="article-tag__name">{ name && name.charAt(0).toUpperCase() + name.slice(1)}</span></h2>
                  { articles.length ? <ArticleListView articles={articles}/>
                    : <div>No article with the tag &lsquo;{ name }&rsquo; was found</div> }
                </div>
              </div>
            </div>
          </div>
        </section>)}
      </React.Fragment>
    );
  }
}

ArticlesByTag.propTypes = {
  loading: PropTypes.bool.isRequired,
  articles: PropTypes.array,
  match: PropTypes.object.isRequired,
  getArticlesByTag: PropTypes.func.isRequired,
  handleLogout: PropTypes.func
};

const mapStateToProps = state => ({
  loading: state.articlesByTag.loading,
  errors: state.articlesByTag.errors,
  articles: state.articlesByTag.articles
});

const mapDispatchToProps = {
  getArticlesByTag,
  handleLogout: logout
};

export default connect(mapStateToProps, mapDispatchToProps)(ArticlesByTag);
