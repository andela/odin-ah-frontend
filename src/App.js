import React from 'react';
import logo from './logo.svg';
import './App.css';
import Todos from './components/Todos.jsx';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo"/>
        <p className='test'>
          Authors Haven react base template
        </p>
        <Todos/>
      </header>
    </div>
  );
}

export default App;
