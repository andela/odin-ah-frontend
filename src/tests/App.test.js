import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';
import { Todos } from '../components/Todos';

it('renders Todos without crashing', () => {
  const wrapper = shallow(<Todos />);
  expect(wrapper).toMatchSnapshot();
});


it('renders without crashing using enzyme', () => {
  shallow(<App/>);
});

it('renders welcome message', () => {
  const wrapper = shallow(<App/>);
  const welcome = <p className='test'>Authors Haven react base template</p>;
  expect(wrapper.contains(welcome))
    .toEqual(true);
});
