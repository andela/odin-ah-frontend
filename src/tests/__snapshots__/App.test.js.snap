// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Todos without crashing 1`] = `
<div>
  <h1>
    Hello I have a state in my redux store
  </h1>
</div>
`;
